ts/case-report.ts

```diff
@@ -0,0 +1,637 @@
+import { AttributeDefinition, AttributeList } from "./component/attribute-list";
+import { TableBuilder, TableDefinition } from "./component/table-builder";
+import { Metric, MetricCategory, MetricSubcategory } from "./data/assay";
+import { Case } from "./data/case";
+import { qcStatuses } from "./data/qc-status";
+import {
+  getMetricValue,
+  getRequisitionMetricCellHighlight,
+  getRequisitionQcStatus,
+  makeRequisitionMetricDisplay,
+  metricApplies as requisitionMetricApplies,
+  Requisition,
+  RequisitionQc,
+  RequisitionQcGroup,
+  subcategoryApplies as requisitionSubcategoryApplies,
+} from "./data/requisition";
+import {
+  addMetricValueContents,
+  getFirstReviewStatus,
+  getSampleMetricCellHighlight,
+  metricApplies as sampleMetricApplies,
+  RUN_METRIC_LABELS,
+  Sample,
+  subcategoryApplies as sampleSubcategoryApplies,
+} from "./data/sample";
+import { addTextDiv, makeNameDiv } from "./util/html-utils";
+import { getMetricRequirementText } from "./util/metrics";
+import { get } from "./util/requests";
+import { siteConfig } from "./util/site-config";
+import { urls } from "./util/urls";
+
+interface ReportSample {
+  sample: Sample;
+  allSamples?: Sample[]; // for run-level subcategories
+  metricCategory: MetricCategory;
+  metricSubcategory: MetricSubcategory;
+  caseAssayId: number;
+}
+
+interface ReportInformatics {
+  requisition: Requisition;
+  requisitionQcGroup: RequisitionQcGroup;
+  requisitionQc: RequisitionQc | null; // null if no informatics review QC has been added
+  metricCategory: MetricCategory;
+  metricSubcategory: MetricSubcategory;
+}
+
+const attributes: AttributeDefinition<Case>[] = [
+  {
+    title: "Case ID",
+    addContents(object, fragment) {
+      fragment.appendChild(
+        makeNameDiv(object.id, undefined, urls.dimsum.case(object.id))
+      );
+    },
+  },
+  {
+    title: "Requisition",
+    addContents(object, fragment) {
+      fragment.appendChild(
+        makeNameDiv(
+          object.requisition.name,
+          urls.miso.requisition(object.requisition.id),
+          urls.dimsum.requisition(object.requisition.id)
+        )
+      );
+    },
+  },
+  {
+    title: "Internal Name",
+    addContents(object, fragment) {
+      fragment.appendChild(
+        makeNameDiv(
+          object.donor.name,
+          urls.miso.sample(object.donor.id),
+          urls.dimsum.donor(object.donor.id)
```

Comment:
Shouldn't the url for dimsum use donor name not the id?

Reply:
Yes. Good catch, thanks! Will fix